osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -89,17 +91,76 @@ private void setDistances(double clockRate)
             if (lastObject is Slider lastSlider)
             {
                 computeSliderCursorPosition(lastSlider);
-                TravelDistance = lastSlider.LazyTravelDistance * scalingFactor;
+                TravelDistance = 0;
                 TravelTime = Math.Max(lastSlider.LazyTravelTime / clockRate, min_delta_time);
                 MovementTime = Math.Max(StrainTime - TravelTime, min_delta_time);
+                MovementDistance = Vector2.Subtract(lastSlider.TailCircle.StackedPosition, BaseObject.StackedPosition).Length * scalingFactor;
+
+                int repeatCount = 0;
+
+                Vector2 currSliderPosition = ((OsuHitObject)lastSlider.NestedHitObjects[0]).StackedPosition;
+
+                for (int i = 1; i < lastSlider.NestedHitObjects.Count; i++)
+                {
+                    var currSliderObj = (OsuHitObject)lastSlider.NestedHitObjects[i];
+
+                    Vector2 currSlider = Vector2.Subtract(currSliderObj.StackedPosition, currSliderPosition);
+                    double currSliderLength = currSlider.Length * scalingFactor;
+
+                    if (currSliderObj is SliderEndCircle && !(currSliderObj is SliderRepeat))
+                    {
+                        // Calculated above/
+                        Debug.Assert(lastSlider.LazyEndPosition != null);
+
+                        Vector2 lazySlider = Vector2.Subtract((Vector2)lastSlider.LazyEndPosition, currSliderPosition);
+                        if (lazySlider.Length < currSlider.Length)
+                            currSlider = lazySlider; // Take the least distance from slider end vs lazy end.
+
+                        currSliderLength = currSlider.Length * scalingFactor;
+                    }
+
+                    if (currSliderObj is SliderTick)
+                    {
+                        if (currSliderLength > minimum_slider_radius) // minimum_slider_radius is used here as 120 = 2.4 * radius, which means that the cursor assumes the position of least movement required to reach the active tick window.
+                        {
+                            currSliderPosition = Vector2.Add(currSliderPosition, Vector2.Multiply(currSlider, (float)((currSliderLength - minimum_slider_radius) / currSliderLength)));
```

Comment:
Why is this only one value? Wouldn't you expect both X and Y to change independently?

Reply:
This is calculating a lazy slider position on the slider tick. It is changing X and Y. You have the X,Y position of the current last cursor position, and then you add to that the vector to the next note, scaled such that the resulting position is 120 distance away from the center tick position.

Perhaps a schematic would help.